data/transactions/logic/evalStateful_test.go

```diff
@@ -1183,20 +1189,71 @@ func TestAssetDisambiguation(t *testing.T) {
 	partitiontest.PartitionTest(t)
 	t.Parallel()
 
-	// start at 4 when the two meanings were added, stop at 8 because 9 removed the dual meaning
-	testLogicRange(t, 4, 8, func(t *testing.T, ep *EvalParams, tx *transactions.Transaction, ledger *Ledger) {
+	// It would be nice to start at 2, when apps were added, but `assert` is
```

Comment:
this function is insane.

Reply:
And that, in a nutshell, is why forbidding low resource numbers is good - it means nobody has to understand these rules.